### DISCUSSION TITLE
Build failed with appDir & firebase firestore

### DISCUSSION THREAD

**Discussion by omarhoumz** [vercel/community#1881] [Help]
### Summary

The build is failing on this NextJs project after I added a Firebase Firestore call in this page:
https://github.com/omarhoumz/aikidoka.ma/blob/main/app/admin/aikidokas/page.tsx

The build is running successfully on "my machine".

Here is the error I'm getting:
<img width="1248" alt="Screenshot 2023-03-26 at 23 19 19" src="https://user-images.githubusercontent.com/40954879/227811133-921fb460-8ae2-4c19-bab3-1957435f03e7.png">

The repo link is public: https://github.com/omarhoumz/aikidoka.ma

Any help would be appreciated.

### Example

https://github.com/omarhoumz/aikidoka.ma

**Comment by leerob** [comment]
Based on the error message, it appears you are missing permissions on the resource you're trying to access for Firebase. Have you checked that out?

**Comment by omarhoumz** [comment]
I was a new Firebase project. The default Firestore production rules are locked 🔒. I adjusted the rules and it deployed successfully. Thanks 🙏

One error I saw in the build logs and didn't understand is this:
<img width="1248" alt="Screenshot 2023-03-27 at 09 23 49" src="https://user-images.githubusercontent.com/40954879/227900331-47399532-ebd5-47d8-8ce5-ac848533c109.png">
